packages/perseus/src/widgets/interactive-graphs/graphs/polygon.test.tsx

```diff
@@ -605,3 +609,51 @@ describe("getSideSnapConstraint", () => {
         });
     });
 });
+
+describe("getAngleSnapConstraint", () => {
+    it("should find the next available coordinate to maintain a whole length sides", () => {
+        const range: PolygonGraphState["range"] = [
+            [-10, 10],
+            [-10, 10],
+        ];
+        const points: PolygonGraphState["coords"] = [
+            [0, 0],
+            [0, 2],
+            [2, 2],
+            [2, 0],
+        ];
+
+        // We're moving the third point in the top right corner of the polygon (square).
+        const constraint = getAngleSnapConstraint(points, 2, range);
+
+        expect(constraint).toEqual({
```

Comment:
Can you leave a quick comment here about what angle these new points are supposed to represent?